yeelight/main.py

```diff
@@ -59,15 +61,25 @@ def _command(f, *args, **kw):
     if result:
         return result[0]
 
+def get_ip_address(ifname):
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    return socket.inet_ntoa(fcntl.ioctl(
+        s.fileno(),
+        0x8915,  # SIOCGIFADDR
+        struct.pack('256s', ifname[:15])
+    )[20:24])
 
-def discover_bulbs(timeout=2):
+def discover_bulbs(timeout=2, interface=False):
     """
     Discover all the bulbs in the local network.
 
     :param int timeout: How many seconds to wait for replies. Discovery will
                         always take exactly this long to run, as it can't know
                         when all the bulbs have finished responding.
 
+    :param string interface: The interface that should be used for multicast packets.
```

Comment:
This should be an IPv4 address, correct? Can you add that to the description?

Reply:
Added clarification in the latest commit.